src/webui/webui.cpp

```diff
@@ -145,7 +146,40 @@ void WebUI::configure()
     }
 }
 
+bool WebUI::isEnabled() const
+{
+    return m_isEnabled;
+}
+
 bool WebUI::isErrored() const
 {
     return m_isErrored;
 }
+
+bool WebUI::isHttps() const
+{
+    if (!m_httpServer) return false;
+    return m_httpServer->isHttps();
+}
+
+QHostAddress WebUI::hostAddress() const
+{
+    if (!m_httpServer) return {};
+    return m_httpServer->serverAddress();
+}
+
+quint16 WebUI::port() const
+{
+    if (!m_httpServer) return 0;
```

Comment:
Port `0` has special meanings so it isn't correct here. I would suggest returning `std::optional<quint16>` instead.

Reply:
The current intention is that all these properties are invalid in case when WebUI is disabled. So why complicate it? You just/still need to check whether WebUI is enabled first.

Reply:
Basically yes, theese values are invalid if the server is down.

And even if they will be options, what path should the code take when one of them turns out to be empty?
The most obvious path will be to fallback to some default value, which effectively leads to the same result as not having the option.